authenticate-oauth/Web/Authenticate/OAuth.hs

```diff
@@ -124,6 +126,7 @@ newOAuth = OAuth { oauthSignatureMethod = HMACSHA1
                  , oauthConsumerKey = error "You MUST specify oauthConsumerKey parameter."
                  , oauthConsumerSecret = error "You MUST specify oauthConsumerSecret parameter."
                  , oauthVersion = OAuth10a
+                 , oauthBodyHashMethod = SHA1
```

Comment:
Not a security expert but should the default here be `SHA256`?

Reply:
Ah, the intent here is to preserve the library's current behavior of hardcoding to SHA-1 while allowing it to be overridden. The body hash is later signed by the `oauthSignatureMethod` so I don't think it's too egregious. I was thinking of a minor version bump, but maybe we should aim a bit differently. As to the default, `oauthSignatureMethod = HMACSHA1` is specified above. The [spec](https://tools.ietf.org/id/draft-eaton-oauth-bodyhash-00.html#parameter) only mentions SHA-1, but was drafted around the time when successors to SHA-1 were coming into existence and does [mention its deprecation](https://tools.ietf.org/id/draft-eaton-oauth-bodyhash-00.html#anchor13). Implementers need to override this anyway depending on what service they're integrating with. I think the preferred solution would be to follow the signature method's hash which the spec does [allude to](https://tools.ietf.org/id/draft-eaton-oauth-bodyhash-00.html#toc). That should be a major version bump, to be fair to users. Curious to see what the library maintainers prefer. I also wonder if there are any integrations which happen to require sha1 body hashing with a different signature hash method. Suppose some additional configuration could be added to support it, if that were that case.

Reply:
I'd guess this is more likely to be merged upstream if it's done in a non-breaking way, and I think changing the default probably doesn't really make a difference to us anyway (because we are going to be explicitly setting it to SHA256), so I'd suggest leaving the default alone (as you've done).

Reply:
All good, thanks for the clarification!